src/ImageSharp.Drawing/Processors/FillShapeProcessor.cs

```diff
@@ -61,77 +62,398 @@ protected override void OnApply(ImageBase<TColor> source, Rectangle sourceRectan
+                                        distance = nextPoint - y;
+                                    }
+                                    else if (lastPoint + AntialiasFactor > y)
+                                    {
+                                        // we are near the right of the line
+                                        distance = y - lastPoint;
+                                    }
+                                    else
+                                    {
+                                        // we are to far away from the line
+                                        continue;
+                                    }
+                                    opacity = 1 - (distance / AntialiasFactor);
+                                }
+                                else
+                                {
+                                    continue;
+                                }
+                            }
+
+                            // don't set full opactiy color as it will have been gotten by the first scan
+                            if (opacity > Constants.Epsilon && opacity < 1)
+                            {
+                                Vector4 backgroundVector = sourcePixels[x, y].ToVector4();
+                                Vector4 sourceVector = applicator.GetColor(currentPoint).ToVector4();
+
+                                Vector4 finalColor = Vector4BlendTransforms.PremultipliedLerp(backgroundVector, sourceVector, opacity);
+                                finalColor.W = backgroundVector.W;
+
+                                TColor packed = default(TColor);
+                                packed.PackFromVector4(finalColor);
+                                sourcePixels[x, y] = packed;
+                            }
+                        }
+
+                        arrayPool.Return(buffer);
+                    });
+                }
             }
         }
 
-        /// <summary>
-        /// Returns the correct alpha value for the given distance.
-        /// </summary>
-        /// <param name="distance">
-        /// The distance.
-        /// </param>
-        /// <returns>
-        /// The <see cref="float"/>.
-        /// </returns>
-        private float Opacity(float distance)
+        private static void Swap(Vector2[] data, int left, int right)
+        {
+            Vector2 tmp = data[left];
+            data[left] = data[right];
+            data[right] = tmp;
+        }
+
+        private static void QuickSortY(Vector2[] data, int size)
+        {
+            QuickSortY(data, 0, size - 1);
+        }
+
+        private static void QuickSortY(Vector2[] data, int lo, int hi)
+        {
+            if (lo < hi)
+            {
+                int p = PartitionY(data, lo, hi);
+                QuickSortY(data, lo, p);
+                QuickSortY(data, p + 1, hi);
+            }
+        }
+
+        private static void QuickSortX(Vector2[] data, int size)
```

Comment:
Cool to have this non-allocating, delegatele-free quicksort!
However, there is some bloated code. @JimBobSquarePants  I think we should consider using T4 for this, or for stuff like `IPackedBytes`. I'm definitely sure, there will be more cases like this!